judge/comments.py

```diff
@@ -72,6 +72,14 @@ def post(self, request, *args, **kwargs):
         if self.is_comment_locked():
             return HttpResponseForbidden()
 
+        parent = request.POST.get('parent')
+        if parent:
+            try:
+                if not Comment.objects.filter(hidden=False, id=int(parent), page=page).exists():
+                    return HttpResponseNotFound()
+            except (ValueError, Comment.DoesNotExist):
```

Comment:
`Comment.DoesNotExist` is no longer a possible exception.

Also suggest that you do `parent = int(parent)` separately, only put that in `try:`, and put the actual `exists` call in the `else` branch of the `try`, for proper coding practice.